### PR TITLE
Gchehade/stream encode decode

### PR DESCRIPTION
**Pull request by poolpOrg**

### CODE REVIEW

**Comment by poolpOrg** [comment]
nope nope nope

**Review by omar-polo — COMMENTED** [review]
I've left some nits, but overall it reads fine to me.

File: compression/compression.go [hunk 1]
```diff
@@ -77,10 +77,12 @@ func DeflateStream(name string, r io.Reader) (io.Reader, error) {
 }
 
 func DeflateGzipStream(r io.Reader) (io.Reader, error) {
+	gw := gzip.NewWriter(nil) // Initialize the writer
 	pr, pw := io.Pipe()
+	gw.Reset(pw) // Reset the writer to use the pipe writer
+
```
[the remark this answers is not in the document]
Comment (poolpOrg) [reply compression/compression.go]: fixed

File: compression/compression.go [hunk 2]
```diff
@@ -144,10 +148,10 @@ func InflateGzipStream(r io.Reader) (io.Reader, error) {
 }
 
 func InflateLZ4Stream(r io.Reader) (io.Reader, error) {
+	lz := lz4.NewReader(r)
 	pr, pw := io.Pipe()
 	go func() {
 		defer pw.Close()
-		lz := lz4.NewReader(r)
```
[the remark this answers is not in the document]
Comment (poolpOrg) [reply compression/compression.go]: fixed

File: compression/compression.go [hunk 3]
```diff
@@ -91,10 +93,12 @@ func DeflateGzipStream(r io.Reader) (io.Reader, error) {
 }
 
 func DeflateLZ4Stream(r io.Reader) (io.Reader, error) {
+	lw := lz4.NewWriter(nil) // Initialize the writer
 	pr, pw := io.Pipe()
+	lw.Reset(pw) // Reset the writer to use the pipe writer
+
 	go func() {
 		defer pw.Close()
-		lw := lz4.NewWriter(pw)
```
[the remark this answers is not in the document]
Comment (poolpOrg) [reply compression/compression.go]: fixed

File: encryption/symmetric.go [hunk 4]
```diff
@@ -105,38 +105,75 @@ func EncryptStream(key []byte, r io.Reader) (io.Reader, error) {
...
+			return
+		}
+		if _, err := pw.Write(encSubkey); err != nil {
+			pw.CloseWithError(err)
+			return
+		}
+
+		// Create a buffer to accumulate input data
+		buffer := make([]byte, 0, chunkSize)
+		tmp := make([]byte, chunkSize)
```
[the remark this answers is not in the document]
Comment (poolpOrg) [reply encryption/symmetric.go]: fixed

File: encryption/symmetric.go [hunk 5]
```diff
@@ -105,38 +105,75 @@ func EncryptStream(key []byte, r io.Reader) (io.Reader, error) {
...
 
-		// Encrypt and write the actual data in chunks
-		buf := make([]byte, chunkSize)
 		for {
-			n, err := r.Read(buf)
+			// Read data into the temporary buffer
+			n, err := r.Read(tmp)
```
[the remark this answers is not in the document]
Comment (poolpOrg) [reply encryption/symmetric.go:131]: indeed, will rework this

[the remark this answers is not in the document]
Comment (poolpOrg) [reply encryption/symmetric.go]: fixed

**Review by omar-polo — APPROVED** [review]
sorry for nitpicking :)

LGTM

**Pull request merged**